lib/segment/src/data_types/index.rs

```diff
@@ -198,6 +198,136 @@ pub struct TextIndexParams {
     // todo(phrase_match): remove skip
     #[serde(skip)]
     pub phrase_matching: Option<bool>,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub stopwords: Option<StopwordsInterface>,
+}
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema, Clone, PartialEq, Hash, Eq)]
+#[serde(untagged)]
+pub enum StopwordsInterface {
+    Language(Language),
+    Set(StopwordsSet),
+}
+
+#[derive(Debug, Serialize, JsonSchema, Clone, PartialEq, Hash, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum Language {
+    #[serde(alias = "unspecified")]
+    UnspecifiedLanguage,
+    #[serde(alias = "ar")]
+    Arabic,
+    #[serde(alias = "az")]
+    Azerbaijani,
+    #[serde(alias = "eu")]
+    Basque,
+    #[serde(alias = "bn")]
+    Bengali,
+    #[serde(alias = "ca")]
+    Catalan,
+    #[serde(alias = "zh")]
+    Chinese,
+    #[serde(alias = "da")]
+    Danish,
+    #[serde(alias = "nl")]
+    Dutch,
+    #[serde(alias = "en")]
+    English,
+    #[serde(alias = "fi")]
+    Finnish,
+    #[serde(alias = "fr")]
+    French,
+    #[serde(alias = "de")]
+    German,
+    #[serde(alias = "el")]
+    Greek,
+    #[serde(alias = "he")]
+    Hebrew,
+    #[serde(alias = "hi-en")]
+    Hinglish,
+    #[serde(alias = "hu")]
+    Hungarian,
+    #[serde(alias = "id")]
+    Indonesian,
+    #[serde(alias = "it")]
+    Italian,
+    #[serde(alias = "kk")]
+    Kazakh,
+    #[serde(alias = "ne")]
+    Nepali,
+    #[serde(alias = "no")]
+    Norwegian,
+    #[serde(alias = "pt")]
+    Portuguese,
+    #[serde(alias = "ro")]
+    Romanian,
+    #[serde(alias = "ru")]
+    Russian,
+    #[serde(alias = "sl")]
+    Slovene,
+    #[serde(alias = "es")]
+    Spanish,
+    #[serde(alias = "sv")]
+    Swedish,
+    #[serde(alias = "tg")]
+    Tajik,
+    #[serde(alias = "tr")]
+    Turkish,
+}
+
+impl<'de> serde::Deserialize<'de> for Language {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        let lowercase_s = s.to_lowercase();
+
+        match lowercase_s.as_str() {
+            "unspecified" | "unspecifiedlanguage" => Ok(Language::UnspecifiedLanguage),
```

Comment:
What's the expected behaviour of `unspecified`? What's the difference with None?

Reply:
I don't agree with this approach. This feels like a gRPC implementation detail that bleeds throughout the rest of our code base.

Let's only keep it in the gRPC API as suggested by the best practice.

In the rest of our code base we can omit that variant, and have only supported languages. Our gRPC front-end can simply disallow the unspecified variant when it gets converted into our internal structures.

If you talk about patch/unset semantics, the user could provide `language: []`.